### PR TITLE
[PAY-1695] DMs: Entrypoint Analytics

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds tracking to entrypoints to DMs

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/models/Analytics.ts [hunk 1]
```diff
@@ -1647,6 +1648,11 @@ type ChatReportUser = {
   reportedUserId: ID
 }
 
+type ChatEntryPoint = {
+  eventName: Name.CHAT_ENTRY_POINT
+  source: 'banner' | 'navmenu' | 'share' | 'profile'
```
Comment (raymondjacobson) [inline packages/common/src/models/Analytics.ts:1653]: no enum, wow!

File: packages/web/src/components/nav/desktop/NavPopupMenu.tsx [hunk 2]
```diff
@@ -62,7 +66,10 @@ const NavPopupMenu = () => {
     isChatEnabled
       ? {
           text: messagesItemText,
-          onClick: () => navigate(CHATS_PAGE),
+          onClick: () => {
+            dispatch(make(Name.CHAT_ENTRY_POINT, { source: 'navmenu' }))
+            navigate(CHATS_PAGE)
```
Comment (raymondjacobson) [inline packages/web/src/components/nav/desktop/NavPopupMenu.tsx:71]: ultra nit, curious for your persp. on. i like the idea of doing "UX" things first, e.g. we shouldn't delay navigation at all costs. in practice i dont think this matters here, but just figured i'd write

**Review by raymondjacobson — APPROVED** [review]
LGTM!!

File: packages/web/src/components/nav/desktop/NavPopupMenu.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/components/nav/desktop/NavPopupMenu.tsx:70]: I usually do the same, this was mistake - updated

File: packages/common/src/models/Analytics.ts (hunk 1 above)
Comment (rickyrombo) [reply packages/common/src/models/Analytics.ts:1653]: my main enum argument is for tracking usages of values, but this seems small enough to search for all `Name.CHAT_ENTRY_POINT` usages

**Pull request merged**